app/controllers/admin/facilities_controller.rb

```diff
@@ -85,13 +94,20 @@ def new_facility(attributes = nil)
 
   def set_facility
     @facility = Facility.friendly.find(params[:id])
-    authorize([:manage, :facility, @facility])
   end
 
   def set_facility_group
     @facility_group = FacilityGroup.friendly.find(params[:facility_group_id])
   end
 
+  def authorize_facility
+    raise Pundit::NotAuthorizedError unless current_admin.can_manage_facilities?(@facility)
```

Comment:
Just piggy-backing on the pundit exception so we can throw up an already wired-up redirect + flash message. We can use our own exceptions later.

Reply:
👍